src/client/src/tscParseLineProtocol.c

```diff
@@ -2077,6 +2078,11 @@ static int32_t parseSmlValue(TAOS_SML_KV *pKV, const char **idx,
     val_rqoute
   } val_state;
 
+  value = malloc(bufSize);
```

Comment:
Use a preallocated buffer in heap instead of a post-allocated one, in order to be able  to construct the result during processing the syntax.

Reply:
Although it is less efficient to use a buffer in heap memory to process short strings than one in stack memory, the old behavior of code is preserved for simplicity since the taosAdapter as a stateless independent app which be horizontally scaled easily on demand. 
According to a basic benchmark of performance with a buffer preallocated v.s. post-allocated , there is not noticeable difference of performance between these two choices.